sync/pkg/server.go

```diff
@@ -131,3 +130,14 @@ func (s *SyncImpl) FetchAllFlags(context.Context, *v1.FetchAllFlagsRequest) (*v1
 		FlagConfiguration: string(marshalled),
 	}, nil
 }
+
+func (s *SyncImpl) GetMetadata(context.Context, *v1.GetMetadataRequest) (*v1.GetMetadataResponse, error) {
+	return &v1.GetMetadataResponse{
+		Metadata: []*v1.KeyValue{
+			{
+				Key:   "paths",
+				Value: fmt.Sprintf("%v", s.fw.paths),
+			},
```

Comment:
I wanted to add _some_ kind of meaningful metadata, even though this is really just a test harness. I chose to send back all the watched flags:

```
{
  "metadata": [
    {
      "key": "paths",
      "value": "[//home/todd/git/flagd-testbed/flags/testing-flags.json //home/todd/git/flagd-testbed/flags/custom-ops.json]"
    }
  ]
}
```

Reply:
Looking at this more, I wonder if we should have used a `map` here.

This key/value construct used to be the recommended way, but maps exist now: https://protobuf.dev/programming-guides/proto3/#maps

Reply:
Good point. Per details in the backward compatibility section, I assume the map is some syntactic sugar for key-value message definitions. And it should make schema easy to understand. 

If this is not a breaking change (per buf validations) I think it's fine to upgrade to map<string, string>.